samples/subsys/ipc/ipm_mhu_dual_core/src/main.c

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) 2019 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <misc/printk.h>
+#include <ipm.h>
+
+enum cpu_id_t {
+	MHU_CPU0 = 0,
+	MHU_CPU1,
+	MHU_CPU_MAX,
+};
+
+/* (Secure System Control) Base Address */
+#define SSE_200_SYSTEM_CTRL_S_BASE	(0x50021000UL)
+#define SSE_200_SYSTEM_CTRL_INITSVTOR1	(SSE_200_SYSTEM_CTRL_S_BASE + 0x114)
+#define SSE_200_SYSTEM_CTRL_CPU_WAIT	(SSE_200_SYSTEM_CTRL_S_BASE + 0x118)
+#define SSE_200_CPU_ID_UNIT_BASE	(0x5001F000UL)
+
+#define NON_SECURE_FLASH_ADDRESS	(192 * 1024)
+#define NON_SECURE_IMAGE_HEADER		(0x400)
+#define NON_SECURE_FLASH_OFFSET		(0x10000000)
+
+struct device *mhu0;
+
+static void wakeup_cpu1(void)
```

Comment:
Why isn't this in the soc or board-level support?

Reply:
@galak Hi Kumar,
What is your opinion on that? The driver would be in ext/hal/xxx folder for each platform, right? Current Zephyr seems no arm folder in ext/hal/.

Reply:
Why would it be in `ext/`? I think it belongs in `soc/arm/arm/musca_a`